metagraph/src/cli/align.cpp

```diff
@@ -405,8 +406,12 @@ int align_to_graph(Config *config) {
                 // Wrap it in CanonicalDBG if needed. This way, each thread gets its
                 // own wrapper (and more importantly, its own local cache).
                 if (aln_graph->get_mode() == DeBruijnGraph::PRIMARY) {
-                    aln_graph = std::make_shared<CanonicalDBG>(aln_graph);
+                    aln_graph = std::make_shared<CanonicalDBG>(aln_graph, 0);
                     logger->trace("Primary graph wrapped into canonical");
+                    // If backwards traversal on DBGSuccinct will be needed, then
+                    // add a cache to speed it up.
+                    if (dbg_succ && aln_graph->get_mode() != DeBruijnGraph::CANONICAL)
+                        aln_graph->add_extension(std::make_shared<NodeFirstCache>(*dbg_succ));
```

Comment:
How do these two caches interact (in `CanonicalDBG` and `NodeFirstCache`)?
Would it make sense to share the same storage of cached elements?

Reply:
From benchmarking, I've found that having a `NodeRC` index or (inclusive) a `NodeFirstCache` leads to better performance and the caches in `CanonicalDBG` actually worsen performance in these cases.

The reason why I want to keep these separate is that `NodeFirstCache` is used by both `CanonicalDBG` and `RCDBG`.

Reply:
So if caching in `CanonicalDBG` doesn't help, why do we keep it?

Reply:
My plan now is to get rid of it, I wanted to run the benchmarks first.

Reply:
anyway, I got rid of it